Update vercel deployment to specify node and npm version

Currently facing an issue where when trying to deploy to vercel we get an error message saying

npm ERR! notsup Required: {"npm":"^9.8.1","node":"^18.[16](https://github.com/ai-cfia/finesse-frontend/actions/runs/7142585744/job/19452590732#step:6:17).0"}
npm ERR! notsup Actual:   {"npm":"10.2.3","node":"v18.19.0"}

to fix this, we will use the "actions/setup-node@v3" and run "npm install -g npm@9.8.1" to specify a npm and node version

showcase and explain in this PR as well:
https://github.com/ai-cfia/finesse-frontend/pull/49

--- a/.github/workflows/workflow-vercel-deployment.yml
+++ b/.github/workflows/workflow-vercel-deployment.yml
@@ -1,7 +1,5 @@
 name: Vercel Static Frontend Deployment
 
-
-
 on:
   workflow_call:
     inputs:
@@ -12,7 +10,6 @@ on:
         required: true
         type: string
         description: 'Deployment environment (dev, prd, uat)'
-
 
 env:
   VERCEL_ORG_ID: ${{ secrets.VERCEL_ORG_ID }}
@@ -29,15 +26,31 @@ jobs:
   deploy:
     runs-on: ubuntu-latest
     steps:
-      - uses: actions/checkout@v3
+
+      - name: checkout code
+        uses: actions/checkout@v3
+
+      - name: install node
+        uses: actions/setup-node@v3
+        with:
+          cache: npm
+          node-version-file: "package.json"
+
+      - name: install npm
+        run: npm install -g npm@9.8.1
+
       - name: Install Vercel CLI
         run: npm install --global vercel@latest
+
       - name: Pull Vercel Environment Information
         run: vercel pull --yes --environment=preview --token=${{ secrets.VERCEL_TOKEN }}
+
       - name: Build Project Artifacts
         run: vercel build --token=${{ secrets.VERCEL_TOKEN }}
+
       - name: Deploy Project Artifacts to Vercel
         run: echo "DEPLOYMENT_URL=$(vercel deploy --prebuilt --token=${{ secrets.VERCEL_TOKEN }})" >> $GITHUB_ENV
+
       - name: Set URL to dev environment
         run: vc alias "$DEPLOYMENT_URL" ${{ env.VERCEL_PROJECT_NAME }}-${{ env.DEPLOYMENT_ENVIRONMENT }}.vercel.app --token=${{ secrets.VERCEL_TOKEN }}
 

